redash/handlers/base.py

```diff
@@ -77,27 +76,15 @@ def get_object_or_404(fn, *args, **kwargs):
     return rv
 
 
-def paginate(query_set, page, page_size, serializer, **kwargs):
-    count = query_set.count()
-
-    if page < 1:
-        abort(400, message="Page must be positive integer.")
-
-    if (page - 1) * page_size + 1 > count > 0:
-        abort(400, message="Page is out of range.")
-
-    if page_size > 250 or page_size < 1:
-        abort(400, message="Page size is out of range (1-250).")
-
-    results = query_set.paginate(page, page_size)
+def paginate(query_set, page, per_page, serializer, **kwargs):
+    results = db.paginate(query_set, page=page, per_page=per_page, max_per_page=250)
```

Comment:
https://github.com/pallets-eco/flask-sqlalchemy/blob/538f0430f51f8e45c20f84d9e3a13512b8077141/src/flask_sqlalchemy/pagination.py#L123
it selects a smaller value from per_page and max_per_page

Reply:
https://github.com/pallets-eco/flask-sqlalchemy/blob/538f0430f51f8e45c20f84d9e3a13512b8077141/src/flask_sqlalchemy/pagination.py#L123
it selects a smaller value from per_page and max_per_page

Reply:
If you specify max_per_page=250, does this mean that even if you specify per_page=251, the result for the 250th page will be returned without an error?

Do you think it's really a good idea to return the results of page=250 when you request data of page=251 instead of returning 400 (Bad Request) error?

I think it would be clearer and better to issue an error in this case. This is because the user may not realize that even though he specified the data for the 251st page, the data for the 250th page is being returned.

What do you think about this?

Reply:
Don't think this is a good idea, but commited it back